src/DotNetWorker/Converters/EnumerableConverter.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker.Converters
+{
+    // Converting IEnumerable<> to Array
+    internal class EnumerableConverter : IConverter
```

Comment:
Adding some suggestions to the class, but given the fact this is pretty specific to gRPC, I'm wondering if this shouldn't be a converted injected by the gRPC integration (mostly thinking about when we break it off) and operating on the gRPC types, which would make this a bit easier and rely less on type checking. We can chat, but wanted to share this to get your thoughts.